lib/content-services/src/lib/security/services/security-controls-groups-marks-security.service.ts

```diff
@@ -0,0 +1,259 @@
+
+    private _reloadAuthorityClearance: Subject<void> = new Subject<void>();
+    reloadAuthorityClearance$ = this._reloadAuthorityClearance.asObservable();
+
+    private marksPaginatedSource = new Subject<SecurityControlsMarkResponse>();
+    marksPaginated$ = this.marksPaginatedSource.asObservable();
+
+    private loadingSource = new BehaviorSubject<boolean>(true);
+    loading$ = this.loadingSource.asObservable();
+
+    private securityGroup: SecurityGroupsApi;
+    private securityMark: SecurityMarksApi;
+    private authorityClearance: AuthorityClearanceApi;
+    constructor(private apiService: AlfrescoApiService) {}
+
+    get groupsApi() {
+        return this.securityGroup || (this.securityGroup = new SecurityGroupsApi(this.apiService.getInstance()));
+    }
+
+    get marksApi() {
+        return this.securityMark || (this.securityMark = new SecurityMarksApi(this.apiService.getInstance()));
+    }
+
+    get authorityClearanceApi() {
+        return this.authorityClearance || (this.authorityClearance = new AuthorityClearanceApi(this.apiService.getInstance()));
+    }
+
+    get reloadAuthorityClearance(): Subject<void> {
+        return this._reloadAuthorityClearance;
+    }
+
+    getSecurityGroup(
+        skipCount = DEFAULT_SKIP_COUNT,
+        maxItems = DEFAULT_MAX_GROUPS,
+        include = DEFAULT_INCLUDE
+    ): Promise<SecurityControlsGroupResponse> {
+        let securityControlsGroupResponse: SecurityControlsGroupResponse;
+        return new Promise((resolve, reject) => {
+            this.groupsApi
+                .getSecurityGroups({
+                    include,
+                    skipCount,
+                    maxItems
+                })
+                .then((response: SecurityGroupPaging) => {
+                    this.groupsPaginatedSource.next(
+                        (securityControlsGroupResponse = {
+                            pagination: response.list.pagination,
+                            entries: response.list.entries.map(
+                                (group: SecurityGroupEntry) => group.entry
+                            )
+                        })
+                    );
+                    resolve(securityControlsGroupResponse);
+                })
+                .catch((error) => reject(error));
+        });
+    }
+
+    createSecurityGroup(
+        input: SecurityGroupBody
+    ): Observable<SecurityGroupEntry> {
+        this.loadingSource.next(true);
+        const payload: SecurityGroupBody = {
+            ...input
+        };
+        const opts = {
+            DEFAULT_INCLUDE
+        };
+        const promise = this.groupsApi.createSecurityGroup(payload, opts);
+
+        return from(promise).pipe(
+            finalize(() => this.loadingSource.next(false))
+        );
+    }
+
+    createSecurityMarks(
+        securityGroupId: string,
+        input: SecurityMarkBody[]
+    ): Promise<any> {
+        this.loadingSource.next(true);
```

Comment:
I see that for some functions (like for that function - createSecurityMarks) it emits true for loading but does not emit false when loading finishes. Should we add for false too?

Reply:
addressed, ty